src/__swaps__/screens/Swap/components/SwapNumberPad.tsx

```diff
@@ -49,23 +51,78 @@ const getFormattedInputKey = (inputKey: inputKeys) => {
 
 export const SwapNumberPad = () => {
   const { isDarkMode } = useColorMode();
-  const { focusedInput, isQuoteStale, SwapInputController, configProgress, outputQuotesAreDisabled } = useSwapContext();
+  const { nativeCurrency } = useAccountSettings();
+  const {
+    focusedInput,
+    internalSelectedInputAsset,
+    internalSelectedOutputAsset,
+    isQuoteStale,
+    SwapInputController,
+    configProgress,
+    outputQuotesAreDisabled,
+  } = useSwapContext();
 
   const longPressTimer = useSharedValue(0);
 
-  const addNumber = (number?: number) => {
+  const removeFormatting = (inputKey: inputKeys) => {
     'worklet';
+    return stripNonDecimalNumbers(SwapInputController[getFormattedInputKey(inputKey)].value);
```

Comment:
created a more generic formatter that strips away commas as well as native currency symbols for native input values